deployment/terraform/destroy.go

```diff
@@ -38,13 +36,13 @@ func (t *Terraform) Destroy() error {
 				"s3://" + t.output.S3Bucket.Id,
 				"--recursive",
 			}
-			if err := t.runAWSCommand(emptyBucketCtx, emptyS3BucketArgs, nil); err != nil {
-				emptyBucketErrCh <- fmt.Errorf("failed to run local cmd \"aws %s\": %w", strings.Join(emptyS3BucketArgs, " "), err)
-				return
-			}
+			// We intentionally ignore potential errors from this command,
+			// since it introduces spurious failures when the bucket is
+			// destroyed before the command finishes.
+			// See https://mattermost.atlassian.net/browse/MM-62075
+			_ = t.runAWSCommand(emptyBucketCtx, emptyS3BucketArgs, nil)
```

Comment:
Is it possible at all to know what error is returned when the bucket is destroyed before the command finishes, and ignore that exact error rather than all errors? If that is too complicated, this is fine as well. Just curious.

Reply:
We could manually grep the returned error, looking for something like `failed to empty`. I don't know if that same string would show up in other errors, though. That would need some more testing.

Reply:
I actually ran into this today. It seems like the error is NoSuchBucket:

```
info  [2024-12-06 14:26:13.837 +05:30] delete failed: s3://agnivalocallt.s3bucket/users/baw7cx384i8ppdka1h1zj3ntdr/profile.png An error occurred (NoSuchBucket) when calling the DeleteObject operation: The specified bucket does not exist caller="terraform/engine.go:71"
info  [2024-12-06 14:26:13.851 +05:30] fatal error: An error occurred (NoSuchBucket) when calling the ListObjectsV2 operation: The specified bucket does not exist caller="terraform/engine.go:71"

```